ipa-core/src/protocol/prss/mod.rs

```diff
@@ -444,28 +415,29 @@ pub mod test {
     /// When inputs are the same, outputs are the same.
     #[test]
     fn generate_equal() {
-        let (g1, g2) = make();
+        const SEED: u64 = 0;
+        let (g1, g2) = make(SEED);
         assert_eq!(g1.generate(0), g2.generate(0));
         assert_eq!(g1.generate(1), g2.generate(1));
-        assert_eq!(g1.generate(u128::MAX), g2.generate(u128::MAX));
+        assert_eq!(g1.generate(u64::MAX), g2.generate(u64::MAX));
 
-        // Calling g1 twice with the same input produces the same output.
-        assert_eq!(g1.generate(12), g1.generate(12));
+        // Calling generators seeded with the same key produce the same output
+        assert_eq!(g1.generate(12), make(SEED).0.generate(12));
         // Now that g1 has been invoked more times than g2, we can check
         // that it isn't cheating by using an internal counter.
         assert_eq!(g1.generate(100), g2.generate(100));
-        assert_eq!(g2.generate(13), g2.generate(13));
+        assert_eq!(g1.generate(13), g2.generate(13));
```

Comment:
@martinthomson I thought that was a typo and you actually wanted to use $G_1$. Let me know if it isn't the case and I can revert this change

Reply:
No, this looks fine.